src/pages/workspace/reportFields/WorkspaceReportFieldsPage.tsx

```diff
@@ -155,7 +155,7 @@ function WorkspaceReportFieldsPage({
 
     const toggleAllReportFields = () => {
         const availableReportFields = Object.values(selectionFieldList).filter((reportField) => reportField.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE);
-        setSelectedReportFields(selectedReportFields.length > 0 ? [] : Object.keys(availableReportFields));
+        setSelectedReportFields(selectedReportFields.length > 0 ? [] : Object.values(availableReportFields).map((reportField) => getReportFieldKey(reportField.fieldID)));
```

Comment:
We should use the object keys as the source of truth. The current solution could break if the object changes (e.g. if we remove `getReportFieldKey` or add more logic). I have already posted a solution https://github.com/Expensify/App/pull/59648#issuecomment-2852652029